src/sql/plot.py

```diff
@@ -605,3 +605,321 @@ def _histogram_stacked(
+    payload,
+    table,
+    column,
+    show_num=False,
+    orient="v",
+    with_=None,
+    conn=None,
+    cmap=None,
+    color=None,
+    edgecolor=None,
+    ax=None,
+):
+    """Plot Bar Chart
+
+    Parameters
+    ----------
+    table : str
+        Table name where the data is located
+
+    column : str
+        Column(s) to plot
+
+    orient : str, default='v'
+        Orientation of the plot. 'v' for vertical and 'h' for horizontal
+
+    conn : connection, default=None
+        Database connection. If None, it uses the current connection
+
+    Notes
+    -----
+
+    .. versionadded:: 0.7.5
+
+    Returns
+    -------
+    ax : matplotlib.Axes
+        Generated plot
+
+    Examples
+    --------
+    .. plot:: ../examples/plot_bar.py
+
+    """
+
+    if not conn:
+        conn = sql.connection.Connection.current
+
+    ax = ax or plt.gca()
+    payload["connection_info"] = conn._get_curr_sqlalchemy_connection_info()
+
+    if column is None:
+        raise ValueError("Column name has not been specified")
+
+    x, height_, xlabel, ylabel = _bar(table, column, with_=with_, conn=conn)
+
+    if color and cmap:
+        # raise a userwarning
+        warnings.warn(
+            "Both color and cmap are given. cmap will be ignored", UserWarning
+        )
+
+    if (not color) and cmap:
+        cmap = plt.get_cmap(cmap)
+        norm = Normalize(vmin=0, vmax=len(x))
+        color = [cmap(norm(i)) for i in range(len(x))]
+
+    if orient == "h":
+        ax.barh(
+            x,
+            height_,
+            align="center",
+            edgecolor=edgecolor,
+            color=color,
+        )
+        ax.set_xlabel(ylabel)
+        ax.set_ylabel(xlabel)
+    else:
+        ax.bar(
+            x,
+            height_,
+            align="center",
+            edgecolor=edgecolor,
+            color=color,
+        )
+        ax.set_ylabel(ylabel)
+        ax.set_xlabel(xlabel)
+
+    if show_num:
+        if orient == "v":
+            for i, v in enumerate(height_):
+                ax.text(
+                    i,
+                    v,
+                    str(v),
+                    color="black",
+                    fontweight="bold",
+                    ha="center",
+                    va="bottom",
+                )
+        else:
+            for i, v in enumerate(height_):
+                ax.text(
+                    v,
+                    i,
+                    str(v),
+                    color="black",
+                    fontweight="bold",
+                    ha="left",
+                    va="center",
+                )
+
+    ax.set_title(table)
+
+    return ax
+
+
+@modify_exceptions
+def _pie(table, column, with_=None, conn=None):
+    """get x and height for bar plot"""
+    if not conn:
+        conn = sql.connection.Connection.current
+    use_backticks = conn.is_use_backtick_template()
+
+    if isinstance(column, list):
+        if len(column) > 2:
+            raise ValueError(
```

Comment:
this should be a usage error: https://jupysql.ploomber.io/en/latest/community/developer-guide.html#throwing-errors